vars/withCamundaOnlyPipeline.groovy

```diff
@@ -115,9 +115,13 @@ def call(type, String product, String component, String s2sServiceName, String t
       onMaster {
         def nonProdEnv = new Environment(env).nonProdName
         def prodEnv = new Environment(env).prodName
+        
         camundaPublish(s2sServiceName, nonProdEnv, product, tenantId)
-        camundaPublish(s2sServiceName, prodEnv, product, tenantId)
-
+        
+        approvedEnvironmentRepository(environment, metricsPublisher) {
```

Comment:
you should be passing a string to this not a class, https://github.com/hmcts/cnp-jenkins-library/blob/task/deploy-to-prod-flag/vars/approvedEnvironmentRepository.groovy#L13